cub/cub/agent/agent_topk.cuh

```diff
@@ -0,0 +1,813 @@
+// SPDX-FileCopyrightText: Copyright (c) 2025, NVIDIA CORPORATION. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+/**
+ * \file
+ * cub::AgentTopK implements a stateful abstraction of CUDA thread blocks for participating in device-wide topK.
+ */
+
+#pragma once
+
+#include <cub/config.cuh>
+
+#if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#  pragma GCC system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#  pragma clang system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#  pragma system_header
+#endif // no system header
+
+#include <cub/block/block_load.cuh>
+#include <cub/block/block_scan.cuh>
+#include <cub/block/block_store.cuh>
+#include <cub/util_type.cuh>
+
+#include <cuda/atomic>
+
+CUB_NAMESPACE_BEGIN
+
+/**
+ * Parameterizable tuning policy type for AgentTopK
+ *
+ * @tparam BlockThreads
+ *   Threads per thread block
+ *
+ * @tparam ItemsPerThread
+ *   Items per thread (per tile of input)
+ *
+ * @tparam BitsPerPass
+ *   Number of bits processed per pass
+ *
+ * @tparam CoefficientForBuffer
+ *   The coefficient parameter for reducing the size of buffer.
+ *   The size of buffer is `1 / CoefficientForBuffer` of original input
+ *
+ * @tparam LoadAlgorithm
+ *   The BlockLoad algorithm to use
+ *
+ * @tparam ScanAlgorithm
+ *   The BlockScan algorithm to use
+ */
+
+template <int BlockThreads,
+          int ItemsPerThread,
+          int BitsPerPass,
+          BlockLoadAlgorithm LoadAlgorithm,
+          BlockScanAlgorithm ScanAlgorithm>
+struct AgentTopKPolicy
+{
+  /// Threads per thread block
+  static constexpr int BLOCK_THREADS = BlockThreads;
```

Comment:
Strong suggestion: Please only add comments for information not already expressed in the code. The variable name is indicative enough here IMO and the comment is not needed. Applies to comments in general (and more instances in this PR).